tokio/src/io/async_fd.rs

```diff
@@ -508,6 +508,92 @@ impl<T: AsRawFd> AsyncFd<T> {
     pub async fn writable_mut<'a>(&'a mut self) -> io::Result<AsyncFdReadyMutGuard<'a, T>> {
         self.readiness_mut(Interest::WRITABLE).await
     }
+
+    /// Reads or writes from the file descriptor using a user-provided IO operation.
+    ///
+    /// This method is useful to retry an operation that might block (by returning
+    /// a [`WouldBlock`] error) until it doesn't block any more.
+    ///
+    /// The readiness of the file descriptor is awaited and when the file descriptor is ready,
+    /// the provided closure is called. The closure should attempt to perform
+    /// IO operation on the file descriptor by manually calling the appropriate syscall.
+    /// If the operation fails because the file descriptor is not actually ready,
+    /// then the closure should return a [`WouldBlock`] error. In such case the
+    /// readiness flag is cleared and the file descriptor readiness is awaited again.
+    /// This loop is repeated until the closure returns an `Ok` or an error
+    /// other than [`WouldBlock`].
+    ///
+    /// The closure should only return a [`WouldBlock`] error if it has performed
+    /// an IO operation on the file descriptor that failed due to the file descriptor not being
+    /// ready. Returning a [`WouldBlock`] error in any other situation will
+    /// incorrectly clear the readiness flag, which can cause the file descriptor to
+    /// behave incorrectly.
+    ///
+    /// The closure should not perform the IO operation using any of the methods
+    /// defined on the Tokio [`AsyncFd`] type, as this will mess with the
+    /// readiness flag and can cause the file descriptor to behave incorrectly.
+    ///
+    /// This method is not intended to be used with combined interests.
+    /// The closure should perform only one type of IO operation, so it should not
+    /// require more than one ready state. This method may panic or sleep forever
+    /// if it is called with a combined interest.
+    ///
+    /// # Examples
+    ///
+    /// This example sends some bytes on the inner [`std::net::UdpSocket`]. [`async_io`] waits
+    /// for readiness, and retries if the send operation does block. This example is equivalent
+    /// to the one given for [`try_io`].
```

Comment:
Please don't start a sentence with something that's lower-case. You'll need to rephrase so that `async_io` isn't the first word.

Reply:
The sentence still starts with "async_io", which is lowercase.

Reply:
turns out I didn't push that change. it's pushed now